### PR TITLE
[Merged by Bors] - feat: exactness of short complexes

### PR DESCRIPTION
**Pull request by joelriou**
This PR introduces the definition of a proposition `S.Exact` when `S` is a short complex.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by jcommelin — COMMENTED** [review]
Under suitable assumptions, this new definition of exactness is equivalent to the existing one in mathlib. I understand that you plan to phase out the old definition. But in the mean time, I think it is good to provide a "glue" lemma that shows that the two are equivalent.

File: Mathlib/Algebra/Homology/ShortComplex/Exact.lean [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  obtain ⟨h, z⟩ := h
+  exact ⟨⟨h.op, (IsZero.of_iso z h.iso.symm).op⟩⟩
+
+lemma Exact.unop {S : ShortComplex Cᵒᵖ} (h : S.Exact) : S.unop.Exact := by
+  obtain ⟨h, z⟩ := h
+  exact ⟨⟨h.unop, (IsZero.of_iso z h.iso.symm).unop⟩⟩
+
+variable (S)
+
+lemma exact_iff_op : S.Exact ↔ S.op.Exact :=
```
[the remark this answers is not in the document]
Comment (joelriou) [reply Mathlib/Algebra/Homology/ShortComplex/Exact.lean]: Yes. I have made this change in my last commit.

**Comment by joelriou** [comment]
Thanks for the suggestions! I shall include a comparison lemma with the current `Exact f g` under suitable assumptions, but I need slightly more basic API (one or two PR on this file) before I can do this.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Comment by joelriou** [comment]
bors retry

**Pull request closed without merging**